File: examples/plugin-with-dependencies/src/main/java/org/intellij/examples/dependencies/plugin/KotlinHelloAction.java
package org.intellij.examples.dependencies.plugin;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.ui.Messages;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.kotlin.idea.KotlinLanguage;

public class KotlinHelloAction extends AnAction {

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        Messages.showInfoMessage("Hello Kotlin Plugin!\n" +
                "I know about " + KotlinLanguage.NAME, "Hello");
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
        e.getPresentation().setIcon(AllIcons.Ide.Notification.InfoEvents);
    }
}

File: examples/kotlin-plugin/src/main/kotlin/org/intellij/examples/kotlin/plugin/KotlinHelloAction.kt
package org.intellij.examples.kotlin.plugin

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent

class KotlinHelloAction : AnAction() {

    override fun actionPerformed(e: AnActionEvent) = Util.sayHello()

    override fun update(e: AnActionEvent) {
        super.update(e)
        e.presentation.icon = AllIcons.General.Information
    }
}
